Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,194 @@
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let SeverityLookup = datatable (severity: int, EventSeverity: string)
+      [
+      0, "High", 
+      1, "High", 
+      2, "High", 
+      3, "Medium",
+      4, "Low",
+      5, "Low", 
+      6, "Informational",
+      7, "Informational" 
+  ];
+  let EventResultWFLookup = datatable (
+      Action_s: string,
+      EventResult_WF: string,
+      DvcAction: string
+  )
+      [
+      "LOG", "Success", "ALLOW",
+      "DENY", "Failure", "DENY",
+      "WARNING", "Success", "WARNING"
+  ];
+  let EventTypeLookup = datatable (LogType_s: string, EventType: string, EventOriginalType: string )
+      [
+      "WF", "HTTPsession", "Web Firewall",
+      "TR", "WebServerSession", "Access"
+  ];
+  let parser=(
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]), 
+      ipaddr_has_any_prefix: dynamic=dynamic([]), 
+      url_has_any: dynamic=dynamic([]),
+      httpuseragent_has_any: dynamic=dynamic([]),
+      eventresult: string='*',
+      disabled: bool=false
+      ) {
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      barracuda_CL
+      | where not(disabled) and (LogType_s == "WF" or LogType_s == "TR")
+      | extend EventStartTime = unixtime_seconds_todatetime(tolong(DeviceReceiptTime_s))
+      | extend EventEndTime = EventStartTime
+      | where (isnull(starttime) or EventStartTime >= starttime)
+          and (isnull(endtime) or EventEndTime <= endtime)
+      | where (array_length(url_has_any) == 0 or URL_s has_any (url_has_any))
+      | where (array_length(httpuseragent_has_any) == 0 or UserAgent_s has_any(httpuseragent_has_any))
+      | extend
+          temp_SrcMatch=has_any_ipv4_prefix(ClientIP_s, src_or_any),
+          temp_DstMatch=has_any_ipv4_prefix(ServerIP_s, ipaddr_has_any_prefix)
+      | extend ASimMatchingIpAddr=case(
+                                  array_length(src_or_any) == 0,
+                                  "-",
+                                  temp_SrcMatch and temp_DstMatch,
+                                  "Both",
+                                  temp_SrcMatch,
+                                  "SrcIpAddr",
+                                  temp_DstMatch,
+                                  "DstIpAddr",
+                                  "No match"
+                              )
+      | where ASimMatchingIpAddr != "No match" 
+      | lookup EventResultWFLookup on Action_s
+      | extend
+          status_code = toint(column_ifexists("HTTPStatus_s", ""))
```

Comment:
Status_code is not mapped to HttpStatusCode and EventResultDetails

Reply:
Done